From 2159fb376e1604b833d4a0b7aa6ea37bc7f48553 Mon Sep 17 00:00:00 2001
From: Tobias <ruck.tobias@gmail.com>
Date: Thu, 18 Jul 2024 19:26:29 +0200
Subject: [PATCH] Update extended keys to use Dogecoin base58 prefixes (#33)

Maintenance rationale (2026-10-14):
Keep Dogecoin extended private/public key serialization interoperable with existing wallet consumers instead of emitting Bitcoin prefixes.

diff --git a/src/kernel/chainparams.cpp b/src/kernel/chainparams.cpp
index 6bd429e9c44f8d7aae7777c9f45ab34753e3f4f3..c6991ae3151705eeafe603f1cebac0155384f8cd 100644
--- a/src/kernel/chainparams.cpp
+++ b/src/kernel/chainparams.cpp
@@ -182,8 +182,8 @@ public:
         base58Prefixes[PUBKEY_ADDRESS] = std::vector<uint8_t>(1, 30);
         base58Prefixes[SCRIPT_ADDRESS] = std::vector<uint8_t>(1, 22);
         base58Prefixes[SECRET_KEY] = std::vector<uint8_t>(1, 158);
-        base58Prefixes[EXT_PUBLIC_KEY] = {0x04, 0x88, 0xB2, 0x1E};
-        base58Prefixes[EXT_SECRET_KEY] = {0x04, 0x88, 0xAD, 0xE4};
+        base58Prefixes[EXT_PUBLIC_KEY] = {0x02, 0xfa, 0xca, 0xfd};
+        base58Prefixes[EXT_SECRET_KEY] = {0x02, 0xfa, 0xc3, 0x98};
         cashaddrPrefix = opts.ecash ? "ecash" : "bitcoincash";
 
         vFixedSeeds = std::vector<SeedSpec6>(std::begin(pnSeed6_main),
diff --git a/src/test/bip32_tests.cpp b/src/test/bip32_tests.cpp
index 76b2b04036a78d73943cd25c6cec049d6348a9ea..1b824e6a0863e4b16c9d2620c544c27708c55c8f 100644
--- a/src/test/bip32_tests.cpp
+++ b/src/test/bip32_tests.cpp
@@ -42,53 +42,53 @@ struct TestVector {
 // clang-format off
 TestVector test1 =
   TestVector("000102030405060708090a0b0c0d0e0f")
-    ("xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjqJoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8",
-     "xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKmPGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi",
+    ("dgub8kXBZ7ymNWy2S8Q3jNgVjFUm5ZJ3QLLaSTdAA89ukSv7Q6MSXwE14b7Nv6eDpE9JJXinTKc8LeLVu19uDPrm5uJuhpKNzV2kAgncwo6bNpP",
+     "dgpv51eADS3spNJh9Gjth94XcPwAczvQaDJs9rqx11kvxKs6r3Ek8AgERHhjLs6mzXQFHRzQqGwqdeoDkZmr8jQMBfi43b7sT3sx3cCSk5fGeUR",
      0x80000000)
-    ("xpub68Gmy5EdvgibQVfPdqkBBCHxA5htiqg55crXYuXoQRKfDBFA1WEjWgP6LHhwBZeNK1VTsfTFUHCdrfp1bgwQ9xv5ski8PX9rL2dZXvgGDnw",
-     "xprv9uHRZZhk6KAJC1avXpDAp4MDc3sQKNxDiPvvkX8Br5ngLNv1TxvUxt4cV1rGL5hj6KCesnDYUhd7oWgT11eZG7XnxHrnYeSvkzY7d2bhkJ7",
+    ("dgub8nnbYqHETn61ajXkw8Z8cHasQNrPnQpb85448DY2ie7PmNecxAm6BjTnhNCvZY3qJk1MKZ9Z5HQasQ83ARb99nmduT7dunvxgcvBFVHuvrq",
+     "dgpv53uaD9MLudRgHssbttwAVS3GwpUkxHnsqUGqy793vX4PDKXvYQDKYS4988T7QEnCzUt7CaGi21e6UKoZnKgXyjna7To1h1aqkcqJBDM65ur",
      1)
-    ("xpub6ASuArnXKPbfEwhqN6e3mwBcDTgzisQN1wXN9BJcM47sSikHjJf3UFHKkNAWbWMiGj7Wf5uMash7SyYq527Hqck2AxYysAA7xmALppuCkwQ",
-     "xprv9wTYmMFdV23N2TdNG573QoEsfRrWKQgWeibmLntzniatZvR9BmLnvSxqu53Kw1UmYPxLgboyZQaXwTCg8MSY3H2EU4pWcQDnRnrVA1xe8fs",
+    ("dgub8pxikcq7rUy5RBaCfPT1D2UXTkqVnSYt4PitiVJqfGubzv9kfyBQ9JN27SfVyUmBGTdQ6ybfBsu4Thrrdkm2qSbaCexVPRwEKMSxYLP2A41",
+     "dgpv565hQvuEJLJk8Kv3d9q36Avw1CTrxKXAmnwgZNurs9rbSs34GCddVzxNYBeB1AZFSZdo1Ps96ibWcGKnufUWkuH1dEkjkmMhRR9fi7Po6B2",
      0x80000002)
-    ("xpub6D4BDPcP2GT577Vvch3R8wDkScZWzQzMMUm3PWbmWvVJrZwQY4VUNgqFJPMM3No2dFDFGTsxxpG5uJh7n7epu4trkrX7x7DogT5Uv6fcLW5",
-     "xprv9z4pot5VBttmtdRTWfWQmoH1taj2axGVzFqSb8C9xaxKymcFzXBDptWmT7FwuEzG3ryjH4ktypQSAewRiNMjANTtpgP4mLTj34bhnZX7UiM",
+    ("dgub8sZzo9eyZMpVHMNHuyrNa2Wfgui23z8sPvxZxpbzq9H3QmLsUj1q3juwfTrLRMCVcyj8iMaGZpU2v319LrJZttkQnYvdUNzv33N6dcqeZ8X",
+     "dgpv58gyTTj61DA9zVi8skEQTAy5EMLPDs7A7LBMoiD232E2riEB4xU4QSWJ6DrnyQ4jx2fBbrp4X8RQqU4YVgPhszifyrKHuhbe2gttLnRB4a6",
      2)
-    ("xpub6FHa3pjLCk84BayeJxFW2SP4XRrFd1JYnxeLeU8EqN3vDfZmbqBqaGJAyiLjTAwm6ZLRQUMv1ZACTj37sR62cfN7fe5JnJ7dh8zL4fiyLHV",
-     "xprvA2JDeKCSNNZky6uBCviVfJSKyQ1mDYahRjijr5idH2WwLsEd4Hsb2Tyh8RfQMuPh7f7RtyzTtdrbdqqsunu5Mm3wDvUAKRHSC34sJ7in334",
+    ("dgub8uoPdamvjqVUMpr1cF4TTXfymizkgaT4qQqsDn8U9aqemryEYViCFKNsLnqiq9ME6HrJrN4DcZN9UTM9S9jmcVDfhLUpJZtk3jGwnGkhd8u",
+     "dgpv5AvNHtr3Bgq94yBra1SVLg8PKAd7rTRMYp4f4fjVMTneDorY8jARc1yDmYGFS4UB1pntDn3dRwsaJexzh6w45PJiP6QPTnRMBfN3rDUiyyH",
      1000000000)
-    ("xpub6H1LXWLaKsWFhvm6RVpEL9P4KfRZSW7abD2ttkWP3SSQvnyA8FSVqNTEcYFgJS2UaFcxupHiYkro49S8yGasTvXEYBVPamhGW6cFJodrTHy",
-     "xprvA41z7zogVVwxVSgdKUHDy1SKmdb533PjDz7J6N6mV6uS3ze1ai8FHa8kmHScGpWmj4WggLyQjgPie1rFSruoUihUZREPSL39UNdE3BBDu76",
+    ("dgub8wXA7GPArxsftAdTindBmEfyZxa4W5G6dfERU4WcMfE9UzNd4uxrWRXvyckfgQRwZz8rMhz29m4k4skAY1EcTkNnZstu73UNrgts2MA5evC",
+     "dgpv5Ce8maTHJpDLbJyJgZ1DeP8P7QCRfxEPM4TDJx7dZYB8vwFvf9R5s88HQQ3TLybFdEC9192aGzQhJpyNEAwnCLxFibAcahB4TzvQbJyp2im",
      0);
 
 TestVector test2 =
   TestVector("fffcf9f6f3f0edeae7e4e1dedbd8d5d2cfccc9c6c3c0bdbab7b4b1aeaba8a5a29f9c999693908d8a8784817e7b7875726f6c696663605d5a5754514e4b484542")
-    ("xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6oDMSgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB",
-     "xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssrdK4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U",
+    ("dgub8kXBZ7ymNWy2RjuNqXknBTCXKkSU5xbQ83QtT4tjiq2yh5Ndi5zwVVGyGCjCXUWGD5xaMzGHjiqkcnt8LamvDpJrZkWqpyXQV4TjDhfyo9Q",
+     "dgpv51eADS3spNJh8tFDoJ8p4bevsC4qFqZgqSdgHxVkvhyy92FwJKTArBsKgvsqB2xLXUjqaZQHukqQr6VxB9o3o32pW1C7bPngcrpg75LUw8V",
      0)
-    ("xpub69H7F5d8KSRgmmdJg2KhpAK8SR3DjMwAdkxj3ZuxV27CprR9LgpeyGmXUbC6wb7ERfvrnKZjXoUmmDznezpbZb7ap6r1D3tgFxHmwMkQTPH",
-     "xprv9vHkqa6EV4sPZHYqZznhT2NPtPCjKuDKGY38FBWLvgaDx45zo9WQRUT3dKYnjwih2yJD9mkrocEZXo1ex8G81dwSM1fwqWpWkeS3v86pgKt",
+    ("dgub8onvpqfirXo6x1VfyK8fFFc3giBinw5ggDAFcsvBoEtwP3pcHMM1eKrDqfh6KZWhRQSkEDG38ogimxJpDjULZQy8qoFWjKfncYaPesrSURc",
+     "dgpv54uuV9jqJP8mf9qWw5Wh8Q4TE9p5xp3yPcP3TmXD17qvpzhusaoF12SaGS9dp6oAw8yfUZp2LvFYCc8mjSJ6jGCDWBcAysxRkGjEUK7pYvw",
      0xFFFFFFFF)
-    ("xpub6ASAVgeehLbnwdqV6UKMHVzgqAG8Gr6riv3Fxxpj8ksbH9ebxaEyBLZ85ySDhKiLDBrQSARLq1uNRts8RuJiHjaDMBU4Zn9h8LZNnBC5y4a",
-     "xprv9wSp6B7kry3Vj9m1zSnLvN3xH8RdsPP1Mh7fAaR7aRLcQMKTR2vidYEeEg2mUCTAwCd6vnxVrcjfy2kRgVsFawNzmjuHc2YmYRmagcEPdU9",
+    ("dgub8pwz5ShFERyD7shrPm8JibHc5TQdLRFNmNEnYGpxSyfKqM44uEmKrPdpT3wD5J7oCvNHt47eS27KSdB9zdxTHZRmNssa63voUvqzVgkMK7p",
+     "dgpv564xjkmMgHJsq23hMXWLbjk1cu2zWJDfUmTaPARyercKHHwNVUDZD6EAsndcYMXeqNJZFb1fPvkedqsYTouEJZdmvuqWkPggY44mEn4uizf",
      1)
-    ("xpub6DF8uhdarytz3FWdA8TvFSvvAh8dP3283MY7p2V4SeE2wyWmG5mg5EwVvmdMVCQcoNJxGoWaU9DCWh89LojfZ537wTfunKau47EL2dhHKon",
-     "xprv9zFnWC6h2cLgpmSA46vutJzBcfJ8yaJGg8cX1e5StJh45BBciYTRXSd25UEPVuesF9yog62tGAQtHjXajPPdbRCHuWS6T8XA2ECKADdw4Ef",
+    ("dgub8skxVTgBQ5GQDVNzTRGsgYDqQzH8ScAe5ojePLVHks1mWAvECkJ2kJ2CHr8LsAp5o6pqihCt59R9XRSAuYPQYttfyA5RJbN1QhWwkCcvPdA",
+     "dgpv58sw9mkHqvc4vdiqRBeuZggExRuVcV8voCxSEE6Jxjxkx7oXnykG6zcYiaqEa4jM9KfFzt63oURrxYehWhRcK3T54gNKbVf51rVViRSkahZ",
      0xFFFFFFFE)
-    ("xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL",
-     "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc",
+    ("dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3",
+     "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP",
      2)
-    ("xpub6FnCn6nSzZAw5Tw7cgR9bi15UV96gLZhjDstkXXxvCLsUXBGXPdSnLFbdpq8p9HmGsApME5hQTZ3emM2rnY5agb9rXpVGyy3bdW6EEgAtqt",
-     "xprvA2nrNbFZABcdryreWet9Ea4LvTJcGsqrMzxHx98MMrotbir7yrKCEXw7nadnHM8Dq38EGfSh6dqA9QWTyefMLEcBYJUuekgW4BYPJcr9E7j",
+    ("dgub8vJ2Mrq3XeYMFhoUuyE72oHzinHbjuiDmg5RKqYCER8c2iajU49oTPLHzuL8C7hEGbgho7n11TkzfVf4RXBpaWShtEDzoFk9xDnhwhVSxT7",
+     "dgpv5BR12Au9yVt1xr9Ksjc8uwkQGDuxungWV5JDAj9DSJ5bUfU34Hc2p5veRhEdMWChjCogbTVrdwr8pDdakxhL3rrxhUR8o7pR3oqZrnPNxDt",
      0);
 
 TestVector test3 =
   TestVector("4b381541583be4423346c643850da4b320e46a87ae3d2a4e6da11eba819cd4acba45d239319ac14f863b8d5ab5a0d0c64d2e8a1e7d1457df2e5a3c51c73235be")
-    ("xpub661MyMwAqRbcEZVB4dScxMAdx6d4nFc9nvyvH3v4gJL378CSRZiYmhRoP7mBy6gSPSCYk6SzXPTf3ND1cZAceL7SfJ1Z3GC8vBgp2epUt13",
-     "xprv9s21ZrQH143K25QhxbucbDDuQ4naNntJRi4KUfWT7xo4EKsHt2QJDu7KXp1A3u7Bi1j8ph3EGsZ9Xvz9dGuVrtHHs7pXeTzjuxBrCmmhgC6",
+    ("dgub8kXBZ7ymNWy2QoMYMvFaPSTZCPmZqpkfqPBSrMvHzX7mfKbuNEEuSkWVkCGBM55uPAiSBz9J8Pfc46X3BHpMe9xzgzR4ZXyFGmyRk9hwu9B",
+     "dgpv51eADS3spNJh7whPKgdcGauxjqPw1hixYnQEhFXKCQ4m7GVCxTh8oT6rAvc184BfcBQb9V6PpBa8Ck7GQawUaWY52Hkknq8euaV2kyg8TB3",
       0x80000000)
-    ("xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBaohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y",
-     "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L",
+    ("dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaYDGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9",
+     "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD",
       0);
 // clang-format on
 
diff --git a/src/test/descriptor_tests.cpp b/src/test/descriptor_tests.cpp
index b29c63fbb1a6c3cc4d0466c9aa36513ad0dea61d..b026f5343f111017736cb1f2a0e5d33f5aae7afe 100644
--- a/src/test/descriptor_tests.cpp
+++ b/src/test/descriptor_tests.cpp
@@ -24,7 +24,7 @@ void CheckUnparsable(const std::string &prv, const std::string &pub,
     auto parse_pub = Parse(pub, keys_pub, error);
     BOOST_CHECK_MESSAGE(!parse_priv, prv);
     BOOST_CHECK_MESSAGE(!parse_pub, pub);
-    BOOST_CHECK(error == expected_error);
+    BOOST_CHECK_EQUAL(error, expected_error);
 }
 
 constexpr int DEFAULT = 0;
@@ -209,7 +209,7 @@ void DoCheck(const std::string &prv, const std::string &pub, int flags,
                     const CPubKey &pk = origin_pair.second.first;
                     BOOST_CHECK(pubkeys.count(pk) > 0);
                 }
-            } else if (pub1.find("xpub") != std::string::npos) {
+            } else if (pub1.find("dgub") != std::string::npos) {
                 // For ranged, hardened derivation, or not ranged, but has an
                 // xpub, all of the keys should appear in the cache
                 BOOST_CHECK(der_xpub_cache.size() + parent_xpub_cache.size() ==
@@ -443,40 +443,40 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
 
     // Versions with BIP32 derivations
     Check("combo([01234567]"
-          "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39"
-          "njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc)",
+          "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7d"
+          "gtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP)",
           "combo([01234567]"
-          "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4"
-          "koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL)",
+          "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLX"
+          "kYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3)",
           SIGNABLE,
           {{"2102d2b36900396c9282fa14628566582f206a5dd0bcc8d5e892611806cafb0301"
             "f0ac",
             "76a91431a507b815593dfc51ffc7245ae7e5aee304246e88ac"}},
           std::nullopt);
     Check("pk("
-          "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7"
-          "AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0)",
+          "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb"
+          "4L41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0)",
           "pk("
-          "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHB"
-          "aohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0)",
+          "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5Rge"
+          "aYDGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0)",
           DEFAULT,
           {{"210379e45b3cf75f9c5f9befd8e9506fb962f6a9d185ac87001ec44a8d3df8d4a9"
             "e3ac"}},
           std::nullopt, {{0}});
     Check("pkh("
-          "xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssr"
-          "dK4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/2147483647'/0)",
+          "dgpv51eADS3spNJh8tFDoJ8p4bevsC4qFqZgqSdgHxVkvhyy92FwJKTArBsKgvsqB2xL"
+          "XUjqaZQHukqQr6VxB9o3o32pW1C7bPngcrpg75LUw8V/2147483647'/0)",
           "pkh("
-          "xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6o"
-          "DMSgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/2147483647'/0)",
+          "dgub8kXBZ7ymNWy2RjuNqXknBTCXKkSU5xbQ83QtT4tjiq2yh5Ndi5zwVVGyGCjCXUWG"
+          "D5xaMzGHjiqkcnt8LamvDpJrZkWqpyXQV4TjDhfyo9Q/2147483647'/0)",
           HARDENED, {{"76a914ebdc90806a9c4356c1c88e42216611e1cb4c1c1788ac"}},
           OutputType::LEGACY, {{0xFFFFFFFFUL, 0}});
     Check("pkh([ffffffff/13']"
-          "xprv9vHkqa6EV4sPZHYqZznhT2NPtPCjKuDKGY38FBWLvgaDx45zo9WQRUT3dKYnjwih"
-          "2yJD9mkrocEZXo1ex8G81dwSM1fwqWpWkeS3v86pgKt/1/2/*)",
+          "dgpv54uuV9jqJP8mf9qWw5Wh8Q4TE9p5xp3yPcP3TmXD17qvpzhusaoF12SaGS9dp6oA"
+          "w8yfUZp2LvFYCc8mjSJ6jGCDWBcAysxRkGjEUK7pYvw/1/2/*)",
           "pkh([ffffffff/13']"
-          "xpub69H7F5d8KSRgmmdJg2KhpAK8SR3DjMwAdkxj3ZuxV27CprR9LgpeyGmXUbC6wb7E"
-          "RfvrnKZjXoUmmDznezpbZb7ap6r1D3tgFxHmwMkQTPH/1/2/*)",
+          "dgub8onvpqfirXo6x1VfyK8fFFc3giBinw5ggDAFcsvBoEtwP3pcHMM1eKrDqfh6KZWh"
+          "RQSkEDG38ogimxJpDjULZQy8qoFWjKfncYaPesrSURc/1/2/*)",
           RANGE,
           {{"76a914326b2249e3a25d5dc60935f044ee835d090ba85988ac"},
            {"76a914af0bd98abc2f2cae66e36896a39ffe2d32984fb788ac"},
@@ -486,11 +486,11 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
            {0x8000000DUL, 1, 2, 1},
            {0x8000000DUL, 1, 2, 2}});
     Check("sh(pkh("
-          "xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKm"
-          "PGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi/10/20/30/40/*'))",
+          "dgpv51eADS3spNJh9Gjth94XcPwAczvQaDJs9rqx11kvxKs6r3Ek8AgERHhjLs6mzXQF"
+          "HRzQqGwqdeoDkZmr8jQMBfi43b7sT3sx3cCSk5fGeUR/10/20/30/40/*'))",
           "sh(pkh("
-          "xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjq"
-          "JoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8/10/20/30/40/*'))",
+          "dgub8kXBZ7ymNWy2S8Q3jNgVjFUm5ZJ3QLLaSTdAA89ukSv7Q6MSXwE14b7Nv6eDpE9J"
+          "JXinTKc8LeLVu19uDPrm5uJuhpKNzV2kAgncwo6bNpP/10/20/30/40/*'))",
           RANGE | HARDENED | DERIVE_HARDENED,
           {{"a9149976cc014a7c012bbc43954a38cdee554865bd1987"},
            {"a914ad29a49cb0420b53d9d6bb8944bcd819c5ff716e87"},
@@ -500,11 +500,11 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
            {10, 20, 30, 40, 0x80000001UL},
            {10, 20, 30, 40, 0x80000002UL}});
     Check("combo("
-          "xprvA2JDeKCSNNZky6uBCviVfJSKyQ1mDYahRjijr5idH2WwLsEd4Hsb2Tyh8RfQMuPh"
-          "7f7RtyzTtdrbdqqsunu5Mm3wDvUAKRHSC34sJ7in334/*)",
+          "dgpv5AvNHtr3Bgq94yBra1SVLg8PKAd7rTRMYp4f4fjVMTneDorY8jARc1yDmYGFS4UB"
+          "1pntDn3dRwsaJexzh6w45PJiP6QPTnRMBfN3rDUiyyH/*)",
           "combo("
-          "xpub6FHa3pjLCk84BayeJxFW2SP4XRrFd1JYnxeLeU8EqN3vDfZmbqBqaGJAyiLjTAwm"
-          "6ZLRQUMv1ZACTj37sR62cfN7fe5JnJ7dh8zL4fiyLHV/*)",
+          "dgub8uoPdamvjqVUMpr1cF4TTXfymizkgaT4qQqsDn8U9aqemryEYViCFKNsLnqiq9ME"
+          "6HrJrN4DcZN9UTM9S9jmcVDfhLUpJZtk3jGwnGkhd8u/*)",
           RANGE,
           {{"2102df12b7035bdac8e3bab862a3a83d06ea6b17b6753d52edecba9be46f5d09e0"
             "76ac",
@@ -516,30 +516,30 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
 
     // Too long key fingerprint
     CheckUnparsable("combo([012345678]"
-                    "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3x"
-                    "z7iAxn8L39njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc)",
+                    "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdW"
+                    "yeMHZdCV7dgtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP)",
                     "combo([012345678]"
-                    "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbm"
-                    "JbZRkrgZw4koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL)",
+                    "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3Gp"
+                    "PHvWFr4YLXkYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3)",
                     "Fingerprint is not 4 bytes (9 characters instead of 8 "
                     "characters)");
     // BIP 32 path element overflow
     CheckUnparsable(
         "pkh("
-        "xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssrdK"
-        "4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/2147483648)",
+        "dgpv51eADS3spNJh8tFDoJ8p4bevsC4qFqZgqSdgHxVkvhyy92FwJKTArBsKgvsqB2xLXU"
+        "jqaZQHukqQr6VxB9o3o32pW1C7bPngcrpg75LUw8V/2147483648)",
         "pkh("
-        "xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6oDM"
-        "Sgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/2147483648)",
+        "dgub8kXBZ7ymNWy2RjuNqXknBTCXKkSU5xbQ83QtT4tjiq2yh5Ndi5zwVVGyGCjCXUWGD5"
+        "xaMzGHjiqkcnt8LamvDpJrZkWqpyXQV4TjDhfyo9Q/2147483648)",
         "Key path value 2147483648 is out of range");
     // Path is not valid uint
     CheckUnparsable(
         "pkh("
-        "xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssrdK"
-        "4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/1aa)",
+        "dgpv51eADS3spNJh8tFDoJ8p4bevsC4qFqZgqSdgHxVkvhyy92FwJKTArBsKgvsqB2xLXU"
+        "jqaZQHukqQr6VxB9o3o32pW1C7bPngcrpg75LUw8V/1aa)",
         "pkh("
-        "xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6oDM"
-        "Sgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/1aa)",
+        "dgub8kXBZ7ymNWy2RjuNqXknBTCXKkSU5xbQ83QtT4tjiq2yh5Ndi5zwVVGyGCjCXUWGD5"
+        "xaMzGHjiqkcnt8LamvDpJrZkWqpyXQV4TjDhfyo9Q/1aa)",
         "Key path value '1aa' is not a valid uint32");
 
     // Multisig constructions
@@ -580,29 +580,29 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "a23552ae"}},
           std::nullopt);
     Check("sh(multi(2,[00000000/111'/222]"
-          "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39"
-          "njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-          "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7"
-          "AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))",
+          "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7d"
+          "gtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+          "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb"
+          "4L41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))",
           "sh(multi(2,[00000000/111'/222]"
-          "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4"
-          "koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-          "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHB"
-          "aohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))",
+          "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLX"
+          "kYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+          "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5Rge"
+          "aYDGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))",
           DEFAULT, {{"a91445a9a622a8b0a1269944be477640eedc447bbd8487"}},
           OutputType::LEGACY, {{0x8000006FUL, 222}, {0}});
     Check("sortedmulti(2,"
-          "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39"
-          "njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc/"
+          "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7d"
+          "gtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP/"
           "*,"
-          "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7"
-          "AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0/0/*)",
+          "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb"
+          "4L41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0/0/*)",
           "sortedmulti(2,"
-          "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4"
-          "koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL/"
+          "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLX"
+          "kYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3/"
           "*,"
-          "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHB"
-          "aohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0/0/*)",
+          "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5Rge"
+          "aYDGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0/0/*)",
           RANGE,
           {{"5221025d5fc65ebb8d44a5274b53bac21ff8307fec2334a32df05553459f8b1f7f"
             "e1b62102fbd47cc8034098f0e6a94c6aeee8528abf0a2153a5d8e46d325b7284c0"
@@ -615,23 +615,23 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
             "fe1f9c52ae"}},
           std::nullopt, {{0}, {1}, {2}, {0, 0, 0}, {0, 0, 1}, {0, 0, 2}});
     Check("sh(multi(2,"
-          "xprv9s21ZrQH143K31xYSDQpPDxsXRTUcvj2iNHm5NUtrGiGG5e2DtALGdso3pGz6ssr"
-          "dK4PFmM8NSpSBHNqPqm55Qn3LqFtT2emdEXVYsCzC2U/2147483647'/"
+          "dgpv51eADS3spNJh8tFDoJ8p4bevsC4qFqZgqSdgHxVkvhyy92FwJKTArBsKgvsqB2xL"
+          "XUjqaZQHukqQr6VxB9o3o32pW1C7bPngcrpg75LUw8V/2147483647'/"
           "0,"
-          "xprv9vHkqa6EV4sPZHYqZznhT2NPtPCjKuDKGY38FBWLvgaDx45zo9WQRUT3dKYnjwih"
-          "2yJD9mkrocEZXo1ex8G81dwSM1fwqWpWkeS3v86pgKt/1/2/"
+          "dgpv54uuV9jqJP8mf9qWw5Wh8Q4TE9p5xp3yPcP3TmXD17qvpzhusaoF12SaGS9dp6oA"
+          "w8yfUZp2LvFYCc8mjSJ6jGCDWBcAysxRkGjEUK7pYvw/1/2/"
           "*,"
-          "xprv9s21ZrQH143K3QTDL4LXw2F7HEK3wJUD2nW2nRk4stbPy6cq3jPPqjiChkVvvNKm"
-          "PGJxWUtg6LnF5kejMRNNU3TGtRBeJgk33yuGBxrMPHi/10/20/30/40/*'))",
+          "dgpv51eADS3spNJh9Gjth94XcPwAczvQaDJs9rqx11kvxKs6r3Ek8AgERHhjLs6mzXQF"
+          "HRzQqGwqdeoDkZmr8jQMBfi43b7sT3sx3cCSk5fGeUR/10/20/30/40/*'))",
           "sh(multi(2,"
-          "xpub661MyMwAqRbcFW31YEwpkMuc5THy2PSt5bDMsktWQcFF8syAmRUapSCGu8ED9W6o"
-          "DMSgv6Zz8idoc4a6mr8BDzTJY47LJhkJ8UB7WEGuduB/2147483647'/"
+          "dgub8kXBZ7ymNWy2RjuNqXknBTCXKkSU5xbQ83QtT4tjiq2yh5Ndi5zwVVGyGCjCXUWG"
+          "D5xaMzGHjiqkcnt8LamvDpJrZkWqpyXQV4TjDhfyo9Q/2147483647'/"
           "0,"
-          "xpub69H7F5d8KSRgmmdJg2KhpAK8SR3DjMwAdkxj3ZuxV27CprR9LgpeyGmXUbC6wb7E"
-          "RfvrnKZjXoUmmDznezpbZb7ap6r1D3tgFxHmwMkQTPH/1/2/"
+          "dgub8onvpqfirXo6x1VfyK8fFFc3giBinw5ggDAFcsvBoEtwP3pcHMM1eKrDqfh6KZWh"
+          "RQSkEDG38ogimxJpDjULZQy8qoFWjKfncYaPesrSURc/1/2/"
           "*,"
-          "xpub661MyMwAqRbcFtXgS5sYJABqqG9YLmC4Q1Rdap9gSE8NqtwybGhePY2gZ29ESFjq"
-          "JoCu1Rupje8YtGqsefD265TMg7usUDFdp6W1EGMcet8/10/20/30/40/*'))",
+          "dgub8kXBZ7ymNWy2S8Q3jNgVjFUm5ZJ3QLLaSTdAA89ukSv7Q6MSXwE14b7Nv6eDpE9J"
+          "JXinTKc8LeLVu19uDPrm5uJuhpKNzV2kAgncwo6bNpP/10/20/30/40/*'))",
           HARDENED | RANGE | DERIVE_HARDENED,
           {{"a914261bbb58cd9714f92e91668d3eed7c4c860f4cdc87"},
            {"a914ff07ad97dc4458ed5236c52bb94ccb7339dedff887"},
@@ -787,123 +787,123 @@ BOOST_AUTO_TEST_CASE(descriptor_test) {
     // Checksums
     Check("sh(multi(2,[00000000/111'/"
           "222]"
-          "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39"
-          "njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-          "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7"
-          "AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))#ggrsrxfy",
+          "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7d"
+          "gtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+          "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb"
+          "4L41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))#5pd0mgtv",
           "sh(multi(2,[00000000/111'/"
           "222]"
-          "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4"
-          "koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-          "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHB"
-          "aohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjg09x5t",
+          "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLX"
+          "kYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+          "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5Rge"
+          "aYDGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))#uzxdm0tx",
           DEFAULT, {{"a91445a9a622a8b0a1269944be477640eedc447bbd8487"}},
           OutputType::LEGACY, {{0x8000006FUL, 222}, {0}});
     Check("sh(multi(2,[00000000/111'/"
           "222]"
-          "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39"
-          "njGVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-          "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7"
-          "AANYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))",
+          "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7d"
+          "gtwxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+          "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb"
+          "4L41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))",
           "sh(multi(2,[00000000/111'/"
           "222]"
-          "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4"
-          "koxb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-          "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHB"
-          "aohPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))",
+          "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLX"
+          "kYUUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+          "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5Rge"
+          "aYDGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))",
           DEFAULT, {{"a91445a9a622a8b0a1269944be477640eedc447bbd8487"}},
           OutputType::LEGACY, {{0x8000006FUL, 222}, {0}});
     // Empty checksum
     CheckUnparsable(
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39nj"
-        "GVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-        "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AA"
-        "NYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))#",
+        "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgt"
+        "wxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+        "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L"
+        "41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))#",
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4ko"
-        "xb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-        "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBao"
-        "hPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#",
+        "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkY"
+        "UUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+        "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaY"
+        "DGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))#",
         "Expected 8 character checksum, not 0 characters");
     // Too long checksum
     CheckUnparsable(
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39nj"
-        "GVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-        "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AA"
-        "NYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))#ggrsrxfyq",
+        "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgt"
+        "wxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+        "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L"
+        "41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))#ggrsrxfyq",
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4ko"
-        "xb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-        "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBao"
-        "hPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjg09x5tq",
+        "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkY"
+        "UUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+        "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaY"
+        "DGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))#tjg09x5tq",
         "Expected 8 character checksum, not 9 characters");
     // Too short checksum
     CheckUnparsable(
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39nj"
-        "GVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-        "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AA"
-        "NYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))#ggrsrxf",
+        "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgt"
+        "wxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+        "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L"
+        "41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))#ggrsrxf",
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4ko"
-        "xb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-        "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBao"
-        "hPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjg09x5",
+        "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkY"
+        "UUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+        "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaY"
+        "DGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))#tjg09x5",
         "Expected 8 character checksum, not 7 characters");
     // Error in payload
     CheckUnparsable(
         "sh(multi(3,[00000000/111'/"
         "222]"
-        "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39nj"
-        "GVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-        "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AA"
-        "NYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))#ggrsrxfy",
+        "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgt"
+        "wxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+        "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L"
+        "41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))#ggrsrxfy",
         "sh(multi(3,[00000000/111'/"
         "222]"
-        "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4ko"
-        "xb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-        "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBao"
-        "hPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjg09x5t",
+        "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkY"
+        "UUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+        "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaY"
+        "DGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))#tjg09x5t",
         "Provided checksum 'tjg09x5t' does not match computed checksum "
-        "'d4x0uxyv'");
+        "'69gdz0mp'");
     // Error in checksum
     CheckUnparsable(
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39nj"
-        "GVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-        "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AA"
-        "NYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))#ggssrxfy",
+        "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgt"
+        "wxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+        "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L"
+        "41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))#ggssrxfy",
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4ko"
-        "xb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-        "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBao"
-        "hPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))#tjq09x4t",
+        "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkY"
+        "UUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+        "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaY"
+        "DGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))#tjq09x4t",
         "Provided checksum 'tjq09x4t' does not match computed checksum "
-        "'tjg09x5t'");
+        "'uzxdm0tx'");
     // Error in checksum
     CheckUnparsable(
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xprvA1RpRA33e1JQ7ifknakTFpgNXPmW2YvmhqLQYMmrj4xJXXWYpDPS3xz7iAxn8L39nj"
-        "GVyuoseXzU6rcxFLJ8HFsTjSyQbLYnMpCqE2VbFWc,"
-        "xprv9uPDJpEQgRQfDcW7BkF7eTya6RPxXeJCqCJGHuCJ4GiRVLzkTXBAJMu2qaMWPrS7AA"
-        "NYqdq6vcBcBUdJCVVFceUvJFjaPdGZ2y9WACViL4L/0))##ggssrxfy",
+        "dgpv5A3y4jgeTKZnDaxS9fUSwCNRsANrfTmRpugKkwnioWE1QU8TtegGdWyeMHZdCV7dgt"
+        "wxJhs3Br1Smfk52eL6zt8EtcudjhghMSW1nDNfmHP,"
+        "dgpv541MxPt1Vjg3KUnnYpy7KqfdSC1KAZ8rxGeBWVDA8hz8NHcfXxTzsutZUgxMU1Wb4L"
+        "41ARtGTvCarHkQyoXELGjhTRfoXzQU2bSgiK5zPHD/0))##ggssrxfy",
         "sh(multi(2,[00000000/111'/"
         "222]"
-        "xpub6ERApfZwUNrhLCkDtcHTcxd75RbzS1ed54G1LkBUHQVHQKqhMkhgbmJbZRkrgZw4ko"
-        "xb5JaHWkY4ALHY2grBGRjaDMzQLcgJvLJuZZvRcEL,"
-        "xpub68NZiKmJWnxxS6aaHmn81bvJeTESw724CRDs6HbuccFQN9Ku14VQrADWgqbhhTHBao"
-        "hPX4CjNLf9fq9MYo6oDaPPLPxSb7gwQN3ih19Zm4Y/0))##tjq09x4t",
+        "dgub8tvzQRcY1UE7WScbBu6R43v2KikVVao97WTXv4BhbdH1xXFAJRE3GpPHvWFr4YLXkY"
+        "UUXCGb7kk1B4bZbRVvGFb8F4PurtTRGvbXH6bMPN3,"
+        "dgub8ntPJ5ou3tLNcLSwb4b5ShDDtkNwzgAaEsRPfbc8vq38vLjMwj1mXDJD3v6h5RgeaY"
+        "DGxwu2yLs6gZTP7XkYDQEwN6Mx7PU3kxLLQaYubK9/0))##tjq09x4t",
         "Multiple '#' symbols");
 
     // Addr and raw tests
